modules/cdc-ext/src/test/java/org/apache/ignite/cdc/CacheConflictOperationsTest.java

```diff
@@ -0,0 +1,312 @@
+
+        putConflict(key, 1, true);
+
+        putConflict(key, 2, true);
+
+        removeConflict(key, 3, true);
+    }
+
+    /**
+     * Tests that {@code IgniteInternalCache#*AllConflict} cache operations works with the conflict resolver
+     * when there is no update conflicts.
+     */
+    @Test
+    public void testUpdatesReorderFromOtherCluster() throws Exception {
+        String key = key("UpdateClusterUpdateReorder", otherClusterId);
+
+        putConflict(key, 2, true);
+
+        // Update with the equal or lower order should fail.
+        putConflict(key, 2, false);
+        putConflict(key, 1, false);
+
+        // Remove with the equal or lower order should fail.
+        removeConflict(key, 2, false);
+        removeConflict(key, 1, false);
+
+        // Remove with the higher order should succeed.
+        putConflict(key, 3, true);
+
+        key = key("UpdateClusterUpdateReorder2", otherClusterId);
+
+        int order = 1;
+
+        putConflict(key, new GridCacheVersion(2, order, 1, otherClusterId), true);
+
+        // Update with the equal or lower topVer should fail.
+        putConflict(key, new GridCacheVersion(2, order, 1, otherClusterId), false);
+        putConflict(key, new GridCacheVersion(1, order, 1, otherClusterId), false);
+
+        // Remove with the equal or lower topVer should fail.
+        removeConflict(key, new GridCacheVersion(2, order, 1, otherClusterId), false);
+        removeConflict(key, new GridCacheVersion(1, order, 1, otherClusterId), false);
+
+        // Remove with the higher topVer should succeed.
+        putConflict(key, new GridCacheVersion(3, order, 1, otherClusterId), true);
+
+        key = key("UpdateClusterUpdateReorder3", otherClusterId);
+
+        int topVer = 1;
+
+        putConflict(key, new GridCacheVersion(topVer, order, 2, otherClusterId), true);
+
+        // Update with the equal or lower nodeOrder should fail.
+        putConflict(key, new GridCacheVersion(topVer, order, 2, otherClusterId), false);
+        putConflict(key, new GridCacheVersion(topVer, order, 1, otherClusterId), false);
+
+        // Remove with the equal or lower nodeOrder should fail.
+        removeConflict(key, new GridCacheVersion(topVer, order, 2, otherClusterId), false);
+        removeConflict(key, new GridCacheVersion(topVer, order, 1, otherClusterId), false);
+
+        // Remove with the higher nodeOrder should succeed.
+        putConflict(key, new GridCacheVersion(topVer, order, 3, otherClusterId), true);
+    }
+
+    /** Tests cache operations for entry replicated from another cluster. */
+    @Test
+    public void testUpdatesConflict() throws Exception {
+        String key = key("UpdateThisClusterConflict0", otherClusterId);
+
+        putConflict(key, 1, true);
+
+        // Local remove for other cluster entry should succeed.
+        remove(key);
+
+        // Conflict replicated update should fail.
```

Comment:
need to explain here that remove makes the comparison by conflictResolveField impossible.